runtime/src/main/java/org/corfudb/recovery/FastObjectLoader.java

```diff
@@ -758,5 +775,30 @@ public void updateLatestCheckpointIfLater(UUID checkPointId) {
             }
         }
     }
+
+    /**
+     * This queue implementation is to be used by single threaded exeuctors
+     * to restrict the amount of pending job submissions.
+     */
+    public class BoundedQueue<E> extends ArrayBlockingQueue<E> {
```

Comment:
Why capacity of ArrayBlockingQueue or LinkedBlockingQueue is not enough for the bounding purpose?

Reply:
Yeah, ADQ / LBQ wouldn't work as you want to block the offers when full. So +1 for the bounded queue. The only thing is that core/max should be the same in the executor which they are.